src/main/java/uk/ac/ebi/eva/evaseqcol/datasource/NCBIAssemblySequencesDataSource.java

```diff
@@ -0,0 +1,107 @@
+package uk.ac.ebi.eva.evaseqcol.datasource;
+
+import org.apache.commons.net.ftp.FTPFile;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.retry.annotation.Backoff;
+import org.springframework.retry.annotation.Retryable;
+import org.springframework.stereotype.Repository;
+
+import uk.ac.ebi.eva.evaseqcol.dus.NCBIAssemblySequencesReader;
+import uk.ac.ebi.eva.evaseqcol.dus.NCBIAssemblySequencesReaderFactory;
+import uk.ac.ebi.eva.evaseqcol.dus.NCBIBrowser;
+import uk.ac.ebi.eva.evaseqcol.dus.NCBIBrowserFactory;
+import uk.ac.ebi.eva.evaseqcol.entities.AssemblySequencesEntity;
+import uk.ac.ebi.eva.evaseqcol.utils.GzipCompress;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.security.NoSuchAlgorithmException;
+import java.util.Optional;
+
+@Repository("NCBISequencesDataSource")
+public class NCBIAssemblySequencesDataSource implements AssemblySequencesDataSource{
```

Comment:
Suggest renaming NCBIAssemblySequencesDataSource to NCBIAssemblySequenceDataSource

Reply:
Okay I'll do it.